### ISSUE TITLE
[FR] Is there any plan to integrate CR-FDM-v2.4.S1.200 board on Creality Sermoon V1?

### ISSUE THREAD

**Issue by duchengyao** [MarlinFirmware/Marlin#24403]
### Describe the feature you want

 Is there any plan to integrate CR-FDM-v2.4.S1.200 board on Creality Sermoon V1?

**Comment by ellensp** [comment]
It already part of this request https://github.com/MarlinFirmware/Configurations/issues/649#issuecomment-1152876729 as you are well aware

**Comment by duchengyao** [comment]
So there is no way to support my sermoon v1, until I get the source code.

**Comment by ellensp** [comment]
You need a pins files that list what each and every pin does.

You can make this yourself, if your a electronics hobbyist or better. You literally track and trace each pin to the CPU to where it goes and document it 

You can also sometimes do this by reverse engineering the firmrware.bin 

Both of these options are last a resort and requires a lot of work

You can also pay someone to do this.Most will require payment in the form of a complete working machine.

**Comment by duchengyao** [comment]
Thanks

**Comment by duchengyao** [comment]
I have created the pin file for SermoonV1.

see https://github.com/Klipper3d/klipper/pull/5621

**Comment by InsanityAutomation** [comment]
Pin file is just the standard 422 board. Nothing different there.

**Comment by duchengyao** [comment]
> Pin file is just the standard 422 board. Nothing different there.

What `standard 422 board` means? Could you share link of the file?

**Comment by ellensp** [comment]
@duchengyao 
ie it the same as pin out as BOARD_CREALITY_V4 in marlin according to InsanityAutomation
https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/Marlin/src/pins/stm32f1/pins_CREALITY_V4.h

**Comment by ellensp** [comment]
we should probably add a motherboard alias for this....

**Comment by duchengyao** [comment]
Really? I think it's definitely different
For example, In sermoon v1, [x_step_pin is PA7](https://github.com/Klipper3d/klipper/pull/5621/files#diff-f14c6fafe05df44cbe9d93498a9ae3fd5059f73e16572067cd71e7429b390c7fR20); In Ender3 and CREALITY_V4 is [PC2](https://github.com/MarlinFirmware/Marlin/blob/bugfix-2.1.x/Marlin/src/pins/stm32f1/pins_CREALITY_V4.h#L101)

**Comment by ellensp** [comment]
I agree these are nothing a like,  @InsanityAutomation  what are you talking about?

**Comment by InsanityAutomation** [comment]
I've got a sermoon D1 and the board in it works with the creality 4.2.2 definition. Board is at the shop, and holiday weekend here in the US so its not in front of me, but I've got quite a few users running the precompiled builds on my GitHub for it. Can check my Creality Dwin2.0 branch and see it just sets the standard board.

**Comment by duchengyao** [comment]
My config file is sermoon `V1` not `D1`,  a very confusing product number.

see https://3dprintbeginner.com/creality-sermoon-v1-pro-review/

**Comment by InsanityAutomation** [comment]
Ah I missed the V in there. Different then!

**Comment by duchengyao** [comment]
Am I allowed to create config file and submit PR?

**Comment by InsanityAutomation** [comment]
Yup, anyone can make a PR!

**Comment by duchengyao** [comment]
official source code has been released: https://github.com/CrealityOfficial/Sermoon-V1
